### PR TITLE
Update build_locally.py for Windows support and compilation mode selection

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR adds Windows OS support for build_locally.py and adds a command line argument `--dubug` to change the compilation mode

### CODE REVIEW

File: scripts/build_locally.py [hunk 1]
```diff
@@ -70,6 +67,14 @@ def run(
         dest="oneapi",
         action="store_true",
     )
+    driver.add_argument(
```
Comment (oleksandr-pavlyk) [inline scripts/build_locally.py:70]: Can we use `choices` keyword for `add_argument` here with default to "Release"?

Comment (vlad-perevezentsev) [reply scripts/build_locally.py:70]: Yes, I will add a list of allowable values for `--debug` argument.

File: scripts/build_locally.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ def run(
         nargs="?",
         const="Debug",
         default="Release",
+        choices=["Release", "Debug"],
```
Comment (oleksandr-pavlyk) [inline scripts/build_locally.py:76]: If the intention was to support `--debug` option with no arguments, the line should be

```python
 driver.add_argument("--debug",
     type=str,
     default="Release",
     const="Debug",
     action="store_const"
  )
```

For example:

```python
In [10]: parser.parse_args([])
Out[10]: Namespace(debug='Release')

In [11]: parser.parse_args(["--debug"])
Out[11]: Namespace(debug='Debug')
```

If the intent is to specify the built type explicitly:

```python
driver.add_argument("--build_type",
    type=str,
    default="Release",
    choices=["Release", "Debug"]
)
```

The usage:

```python
In [15]: parser.parse_args([])
Out[15]: Namespace(debug='Release', build_type='Release')

In [16]: parser.parse_args(["--build_type=Debug"])
Out[16]: Namespace(debug='Release', build_type='Debug')
```

Comment (vlad-perevezentsev) [reply scripts/build_locally.py:76]: Thanks for these examples. I wanted to implement support  `--debug` option without arguments.

Comment (oleksandr-pavlyk) [reply scripts/build_locally.py:76]: Ok, great. Then please make the change. Notice that `narg` parameter is no needed to accomplish that.

**Review by oleksandr-pavlyk — DISMISSED** [review]

File: scripts/build_locally.py [hunk 3]
```diff
@@ -5,25 +5,22 @@
 
 def run(
     use_oneapi=True,
+    build_type="Relese",
```
Comment (diptorupd) [inline scripts/build_locally.py:8]: Typo

Comment (oleksandr-pavlyk) [inline scripts/build_locally.py:8]:
```suggestion
    build_type="Release",
```

**Pull request merged**